development/image-detector/internal/kubernetes/kubernetes_test.go

```diff
@@ -0,0 +1,208 @@
+    app.kubernetes.io/component: dashboard
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/name: dashboard
+    app.kubernetes.io/part-of: tekton-dashboard
+    app.kubernetes.io/version: v0.34
+    dashboard.test.dev/release: v0.34
+    version: v0.34.0
+  name: tekton-dashboard
+  namespace: tekton-pipelines
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: dashboard
+      app.kubernetes.io/instance: default
+      app.kubernetes.io/name: dashboard
+      app.kubernetes.io/part-of: tekton-dashboard
+  template:
+    metadata:
+      labels:
+        app: tekton-dashboard
+        app.kubernetes.io/component: dashboard
+        app.kubernetes.io/instance: default
+        app.kubernetes.io/name: dashboard
+        app.kubernetes.io/part-of: tekton-dashboard
+        app.kubernetes.io/version: v0.34.0
+      name: tekton-dashboard
+    spec:
+      containers:
+        - args:
+            - --port=1000
+          image: gcr.io/tekton-releases/github.com/tektoncd/dashboard/cmd/dashboard:v0.34.0@sha256:3b62e3d2423d28200f4125852c89b23a0725b89af355c4768d60d45d8c30fc47
+`,
+		},
+		{
+			Name:           "complex file with multiple resources, pass",
+			WantErr:        false,
+			ExpectedImages: []string{"test-image:test"},
+			FileContent: `apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: default
+  name: "test"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+  labels:
+    app: test
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+       app: test
+  template:
+    metadata:
+      labels:
+          app: test
+    spec:
+      containers:
+        - name: test
+          image: test-image:test`,
+		},
+		{
+			Name:           "complex file with multiple deployments, pass",
+			WantErr:        false,
+			ExpectedImages: []string{"test-image:test", "test-image:test2"},
+			FileContent: `apiVersion: v1
+kind: ServiceAccount
+metadata:
+  namespace: default
+  name: "test"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+  labels:
+    app: test
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+       app: test
+  template:
+    metadata:
+      labels:
+          app: test
+    spec:
+      containers:
+        - name: test
+          image: test-image:test
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test2-deployment
+  labels:
+    app: test2
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+       app: test2
+  template:
+    metadata:
+      labels:
+          app: test2
+    spec:
+      containers:
+        - name: test2
+          image: test-image:test2`,
+		},
+	}
+
+	for _, c := range tc {
+		t.Run(c.Name, func(t *testing.T) {
+			images, err := extract(strings.NewReader(c.FileContent))
+			if err != nil && !c.WantErr {
+				t.Errorf("Got unexpected error: %v", err)
+			}
+
+			if !reflect.DeepEqual(images, c.ExpectedImages) {
+				t.Errorf("%v != %v", images, c.ExpectedImages)
```

Comment:
This error message could be more descriptive to give a developer more context and details why this test failed. This will be helpful for someone not familiar with the tools and problem domain.

Reply:
I rewrote this type of error messages as well as checked other ones. 
It should be more descriptive now